little_raft/src/replica.rs

```diff
@@ -364,19 +364,20 @@ where
     fn load_new_transitions(&mut self) {
         // Load new transitions. Ignore the transitions if the replica is not
         // the Leader.
+        if self.state != State::Leader {
+            return
+        }
         let transitions = self.state_machine.lock().unwrap().get_pending_transitions();
```

Comment:
I think the replica should be reading pending transitions even if it isn't a leader. Otherwise transitions can get queued up and then get processed by replica once it becomes a leader. The better behavior is to always read pending transitions and then mark them as abandoned if the replica isn't a leader. See https://github.com/andreev-io/little-raft/issues/13.

Reply:
Oh, good point! I didn't consider this case. Yes, replicas definitely need to signal abandoned transitions so that client commands can be restarted or errored out.

Reply:
@suzaku Would you be interested in picking up https://github.com/andreev-io/little-raft/issues/13 as part of this PR?

Reply:
Isn't that one already assigned to someone else?

Reply:
I don't think that person is working on it.

Reply:
OK, I'll take a look at the issue first and see if I understand little-raft enough to be able to work on it.